com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/RenameNewAppsStep.java

```diff
@@ -0,0 +1,70 @@
+package com.sap.cloud.lm.sl.cf.process.steps;
+
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMta;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaModule;
+import com.sap.cloud.lm.sl.cf.process.message.Messages;
+import org.cloudfoundry.client.lib.CloudControllerClient;
+import org.cloudfoundry.client.lib.domain.CloudApplication;
+import org.flowable.engine.delegate.DelegateExecution;
+import org.flowable.variable.api.delegate.VariableScope;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Scope;
+
+import javax.inject.Named;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Named("renameNewAppsStep")
+@Scope(BeanDefinition.SCOPE_PROTOTYPE)
+public class RenameNewAppsStep extends RenameAppsStep {
+
+    @Override
+    protected StepPhase executeStepInternal(ExecutionWrapper execution) {
+        getStepLogger().debug("Renaming new applications");
+
+        CloudControllerClient client = execution.getControllerClient();
+        List<CloudApplication> appsToProcess = getApplicationsToProcess(execution.getContext(), client);
+
+        for (CloudApplication app : appsToProcess) {
+            getStepLogger().debug("Renaming application " + app.getName());
+            client.updateApplicationName(app.getName(), removeAppNameSuffix(app.getName()));
+        }
+        return StepPhase.DONE;
+    }
+
+    private List<CloudApplication> getApplicationsToProcess(VariableScope context, CloudControllerClient client) {
+        DeployedMta deployedMta = StepsUtil.getDeployedMta(context);
+        List<CloudApplication> appsToProcess = client.getApplications();
+        if (deployedMta == null) {
```

Comment:
Actually, I don't understand this entire method. Shouldn't we just rename the apps from `StepUtils.getAppsToDeploy()`?

Reply:
Actually, yes. This is a great idea. I didn't know about this method.